test/libyul/yulSyntaxTests/zkevm/code_source.yul

```diff
@@ -0,0 +1,6 @@
+{
+	let a := code_source()
+}
+// ====
+// dialect: evm
```

Comment:
Could you please check where this value is used?

Reply:
test/libyul/SyntaxTest.cpp reads this value using `string dialectName = m_reader.stringSetting("dialect", "evmTyped");` in the SyntaxTest ctor. (test/TestCaseReader.cpp is responsible for parsing these settings)

The selected dialect is used in yul::test::parse() in test/libyul/Common.cpp which passes it to the main yul parsers and analyzers.

Reply:
So we can introduce a zkevm dialect, taking into account your comment above?

Reply:
We can derive the zkevm dialect from EVMDialect like how EVMDialectTyped is
defined. We can add this to YulStack.cpp's languageToDialect() by defining a
corresponding YulStack::Language. I think the yul optimizations will work
correctly since most of them use the base `Dialect` APIs that are common to all
dialects.

The IRGenerator and what it uses like YulUtilFunctions are hardcoded to generate
EVMDialect (i.e. YulStack::Language::StrictAssembly). I don't see any option to
control the IRGenerator's target dialect. But Dialect translation seems doable
(YulStack::translate()). I don't see any motivation in translating EVMDialect to
our dialect since we're only adding some extensions and not changing the core
EVMDialect yet.

Directly generating our dialect seems doable if we change the IRGenerator's
default behaviour, but I don't know how well tested that path is. So I guess we
can think about this again when we start adding more extensions.